src/jit/gentree.h

```diff
@@ -1078,6 +1076,17 @@ struct GenTree
         }
     }
 
+    bool OperIs(genTreeOps oper)
+    {
+        return OperGet() == oper;
+    }
+
+    template <typename... T>
+    bool OperIs(genTreeOps oper, T... rest)
```

Comment:
Yes, I checked with up to 8 arguments and VC++ doesn't have any problem inlining everything, even when compiling with /O1. I don't know what Clang does but I don't expect it to do anything less.

Note that it isn't recursive. `f(1,2,3)` calls `f(2,3)` which then calls `f(3)`, these are 3 different functions.